timesketch/api/v1/resources/graph.py

```diff
@@ -34,19 +39,90 @@ class GraphListResource(resources.ResourceMixin, Resource):
     """Resource to get all graphs."""
 
     @login_required
-    def get(self):
+    def get(self, sketch_id):
         """Handles GET request to the resource.
 
         Returns:
             List of graphs in JSON (instance of flask.wrappers.Response)
         """
-        graphs = manager.GraphManager.get_graphs()
-        graph_names = {name: graph_class().DISPLAY_NAME for name, graph_class in
-                       graphs}
-        return jsonify(graph_names)
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, 'No sketch found with this ID.')
+
+        graphs = sketch.graphs
+        return self.to_json(graphs)
+
+    def post(self, sketch_id):
+        """Handles POST request to the resource."""
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, 'No sketch found with this ID.')
+
+        form = request.json
+        name = form.get('name')
+        elements = form.get('elements')
+
+        graph = Graph(
+            user=current_user, sketch=sketch, name=str(name),
+            graph_elements=json.dumps(elements))
+        db_session.add(graph)
+        db_session.commit()
+
+        return self.to_json(graph)
 
 
 class GraphResource(resources.ResourceMixin, Resource):
+    """Resource to get all graphs."""
+
+    @login_required
+    def get(self, sketch_id, graph_id):
+        """Handles GET request to the resource.
+
+        Returns:
+            List of graphs in JSON (instance of flask.wrappers.Response)
+        """
+        sketch = Sketch.query.get_with_acl(sketch_id)
```

Comment:
missing a check for sketch here... I would move the `if not sketch` check immediately after this, bail out early if there is no sketch

Reply:
fixed.